pkg/cmd/agent-task/list/list.go

```diff
@@ -0,0 +1,150 @@
+package list
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cli/cli/v2/internal/gh"
+	"github.com/cli/cli/v2/internal/tableprinter"
+	"github.com/cli/cli/v2/pkg/cmd/agent-task/capi"
+	"github.com/cli/cli/v2/pkg/cmd/pr/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+const defaultLimit = 30
+
+// ListOptions are the options for the list command
+type ListOptions struct {
+	IO         *iostreams.IOStreams
+	Config     func() (gh.Config, error)
+	Limit      int
+	CapiClient capi.CapiClient
+	HttpClient func() (*http.Client, error)
+}
+
+// NewCmdList creates the list command
+func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Command {
+	opts := &ListOptions{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		Limit:      defaultLimit,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List agent tasks",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := f.Config()
+			if err != nil {
+				return err
+			}
+
+			httpClient, err := opts.HttpClient()
+			if err != nil {
+				return err
+			}
+
+			authCfg := cfg.Authentication()
+			opts.CapiClient = capi.NewCAPIClient(httpClient, authCfg)
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return listRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func listRun(opts *ListOptions) error {
+	if opts.Limit <= 0 {
+		opts.Limit = defaultLimit
+	}
+
+	capiClient := opts.CapiClient
+
+	opts.IO.StartProgressIndicatorWithLabel("Fetching agent tasks...")
+	defer opts.IO.StopProgressIndicator()
+	sessions, err := capiClient.ListSessionsForViewer(context.Background(), opts.Limit)
+	if err != nil {
+		return err
+	}
+	opts.IO.StopProgressIndicator()
+
+	if len(sessions) == 0 {
+		fmt.Fprintln(opts.IO.Out, "no agent tasks found")
+		return nil
+	}
+
+	cs := opts.IO.ColorScheme()
+	tp := tableprinter.New(opts.IO, tableprinter.WithHeader("Session ID", "Pull Request", "Repo", "Session State", "Created"))
+	for _, s := range sessions {
+		pr := ""
+		if s.ResourceType == "pull" && s.PullRequest.Number != 0 {
+			pr = fmt.Sprintf("#%d", s.PullRequest.Number)
+		} else {
+			// Skip these sessions in case they happen, for now.
+			continue
+		}
+		repo := ""
+		if s.PullRequest.Repository != nil && s.PullRequest.Repository.NameWithOwner != "" {
+			repo = s.PullRequest.Repository.NameWithOwner
+		} else {
+			// Skip these sessions in case they happen, for now.
+			continue
+		}
```

Comment:
How about simplifying these `if`-statements like this?

```suggestion
		if s.ResourceType != "pull" || s.PullRequest == nil || s.PullRequest.Repository == nil {
			// Skip these sessions in case they happen, for now.
			continue
		}

		pr := fmt.Sprintf("#%d", s.PullRequest.Number)
		repo := s.PullRequest.Repository.NameWithOwner
```

I think it's a legitimate expectation to have non-zero/non-empty fields on `s.PullRequest` or `s.PullRequest.NameWithOwner`, and we should only check if they're not `nil` (just for the sake of language semantics).

BTW, if you want to keep it as is, I think we should at least add `s.PullRequest != nil` to both `if`s.

Reply:
Thank you, I like it. Done in 0a5b78a510a7ed44ea833fb3e3da784e0243287d